helm/vertical-pod-autoscaler-app/templates/_helpers.tpl

```diff
@@ -1,65 +1,38 @@
+{{/* vim: set filetype=mustache: */}}
 {{/*
 Expand the name of the chart.
 */}}
-{{- define "vpa.name" -}}
-{{- default .Chart.Name .Values.nameOverride | trunc 63 | trimSuffix "-" }}
-{{- end }}
+{{- define "vertical-pod-autoscaler.name" -}}
+{{- $default := "vertical-pod-autoscaler-app" -}}
+{{- coalesce .Values.nameOverride $default | trunc 63 | trimSuffix "-" -}}
+{{- end -}}
 
 {{/*
 Create a default fully qualified app name.
 We truncate at 63 chars because some Kubernetes name fields are limited to this (by the DNS naming spec).
 If release name contains chart name it will be used as a full name.
 */}}
-{{- define "vpa.fullname" -}}
-{{- if .Values.fullnameOverride }}
-{{- .Values.fullnameOverride | trunc 63 | trimSuffix "-" }}
-{{- else }}
-{{- $name := default .Chart.Name .Values.nameOverride | trimSuffix "-app" }}
-{{- if contains $name .Release.Name }}
-{{- .Release.Name | trunc 63 | trimSuffix "-" }}
-{{- else }}
-{{- printf "%s-%s" .Release.Name $name | trunc 63 | trimSuffix "-" }}
-{{- end }}
-{{- end }}
-{{- end }}
-
-{{/*
-Create chart name and version as used by the chart label.
-*/}}
-{{- define "vpa.chart" -}}
-{{- printf "%s-%s" .Chart.Name .Chart.Version | replace "+" "_" | trunc 63 | trimSuffix "-" }}
-{{- end }}
+{{- define "vertical-pod-autoscaler.fullname" -}}
+{{- if .Values.fullnameOverride -}}
```

Comment:
You look for `fullnameOverride` at the root of values, but it is used in templates in the subchart, which will expect this value in `vertical-pod-autoscaler.fullnameOverride`.
Maybe that's why it does not work ;)

Reply:
Fixed